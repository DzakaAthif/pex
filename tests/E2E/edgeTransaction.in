BUY 0 Avocado 99999 99999;
SELL 1 Avocado 99999 99999;
BUY 2 Wine 1 1;
SELL 3 Wine 1 1;