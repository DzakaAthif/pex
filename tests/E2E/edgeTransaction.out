[PEX] Starting
[PEX] Trading 2 products: Wine Avocado
[PEX] Created FIFO /tmp/pe_exchange_0
[PEX] Created FIFO /tmp/pe_trader_0
[PEX] Starting trader 0 (./trader1)
[PEX] Connected to /tmp/pe_exchange_0
[PEX] Connected to /tmp/pe_trader_0
[PEX] [T0] Parsing command: <BUY 0 Avocado 99999 99999>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 1; Sell levels: 0
[PEX]		BUY 99999 @ $99999 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 1 Avocado 99999 99999>
[PEX] Match: Order 0 [T0], New Order 1 [T0], value: $9999800001, fee: $99998000.
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($-99998000)
[PEX] [T0] Parsing command: <BUY 2 Wine 1 1>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 1; Sell levels: 0
[PEX]		BUY 1 @ $1 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($-99998000)
[PEX] [T0] Parsing command: <SELL 3 Wine 1 1>
[PEX] Match: Order 2 [T0], New Order 3 [T0], value: $1, fee: $0.
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($-99998000)
[PEX] Trader 0 disconnected
[PEX] Trading completed
[PEX] Exchange fees collected: $99998000
